From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Disable network ID generation

Prevents Firefox from generating an internal network ID that could potentially be used to track users.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/netwerk/system/android/nsAndroidNetworkLinkService.cpp b/netwerk/system/android/nsAndroidNetworkLinkService.cpp
index 309115e34d4b..592eda3584fc 100644
--- a/netwerk/system/android/nsAndroidNetworkLinkService.cpp
+++ b/netwerk/system/android/nsAndroidNetworkLinkService.cpp
@@ -126,11 +126,11 @@ nsAndroidNetworkLinkService::GetLinkType(uint32_t* aLinkType) {
 
 NS_IMETHODIMP
 nsAndroidNetworkLinkService::GetNetworkID(nsACString& aNetworkID) {
-  if (!mNetlinkSvc) {
+/*  if (!mNetlinkSvc) {
     return NS_ERROR_NOT_AVAILABLE;
-  }
+  } */
 
-  mNetlinkSvc->GetNetworkID(aNetworkID);
+  aNetworkID.Truncate();
   return NS_OK;
 }
 
diff --git a/netwerk/system/netlink/NetlinkService.cpp b/netwerk/system/netlink/NetlinkService.cpp
index 8941427b3a19..e7981c87b476 100644
--- a/netwerk/system/netlink/NetlinkService.cpp
+++ b/netwerk/system/netlink/NetlinkService.cpp
@@ -1884,8 +1884,8 @@ void NetlinkService::CalculateNetworkID() {
 }
 
 void NetlinkService::GetNetworkID(nsACString& aNetworkID) {
-  MutexAutoLock lock(mMutex);
-  aNetworkID = mNetworkId;
+//  MutexAutoLock lock(mMutex);
+  aNetworkID.Truncate();
 }
 
 nsresult NetlinkService::GetDnsSuffixList(nsTArray<nsCString>& aDnsSuffixList) {
-- 
